### PR TITLE
nspawn,cgtop: make sure --version, --help always work

### PR DESCRIPTION
**Pull request by keszybz**
From #5704.

### CODE REVIEW

File: src/cgtop/cgtop.c [hunk 1]
```diff
@@ -852,7 +852,7 @@ static int parse_argv(int argc, char *argv[]) {
                         }
 
                         arg_recursive = r;
-                        recursive_unset = r == 0;
+                        arg_recursive_unset = r == 0;
```
Comment (poettering) [inline src/cgtop/cgtop.c:855]: hmm, why do we need both arg_recursive and arg_recursive_unset as public fields? Not following? Wouldn't one suffice?

Comment (keszybz) [reply src/cgtop/cgtop.c:855]: I just move the code around. Before it was in a single function, so a stack variable was enough, and now it needs to be accessed from two places, so I made it global.

The code differentiates the case when arg_recursive was explicitly set.

**Comment by keszybz** [comment]
```
ValueError: No JSON object could be decoded
```

**Comment by mbiebl** [comment]
@poettering do you have any more concerns or can this code be merged? It would be good to get #5831 solved so the CI can run the test suite successfully again.

**Pull request merged**